templates/ray-data-llm/README.md

```diff
@@ -0,0 +1,162 @@
+
+
+```python
+from ray.data.llm import vLLMEngineProcessorConfig
+from util.utils import is_on_gcp_cloud
+
+# There's no a10g on GCP.
+accelerator_type = "L4" if is_on_gcp_cloud() else "A10G"
+
+processor_config = vLLMEngineProcessorConfig(
+    model_source="meta-llama/Meta-Llama-3.1-8B-Instruct",
+    engine_kwargs=dict(
+        tensor_parallel_size=1,
+        pipeline_parallel_size=1,
+        max_model_len=16384,
+        enable_chunked_prefill=True,
+        max_num_batched_tokens=2048,
+    ),
+    # Override Ray's runtime env to include the Hugging Face token. Ray is being used under the hood to orchestrate the inference pipeline.
+    runtime_env=dict(
+        env_vars=dict(
+            HF_TOKEN=HF_TOKEN,
+        ),
+    ),
+    batch_size=16,
+    accelerator_type=accelerator_type,
+)
+
+```
+
+## Step 3: Define the preprocess and postprocess lambda
+
+
+We will need to define the preprocess lambda to convert input dataset to format that `vLLMEngineProcessor` can consume, and also postprocessor
+lambda that filter out the uninterested fields from vLLM engine.
+
+
+```python
+preprocess = lambda row: dict(
+    messages=[
+        {
+            "role": "system",
+            "content": "You are a commentator. Your task is to "
+            "summarize highlights from article.",
+        },
+        {
+            "role": "user",
+            "content": f"# Article:\n{row['article']}\n\n"
+            "#Instructions:\nIn clear and concise language, "
+            "summarize the highlights presented in the article.",
+        },
+    ],
+    sampling_params=dict(
+        temperature=0.3,
+        max_tokens=150,
+        detokenize=False,
+    ),
+)
+postprocess = lambda row: {
+    "resp": row["generated_text"],
+}
+```
+
+## Step 4: Build and run the processor
+
+
+With the workload and configs defined, we can now build then run the processor
+
+
+```python
+from ray.data.llm import build_llm_processor
+
+processor = build_llm_processor(
+    processor_config,
+    preprocess=preprocess,
+    postprocess=postprocess,
+)
+
+ds = processor(ds)
+ds = ds.materialize()
+
+
+# Peak the first 3 entries. 
+sampled = ds.take(3)
+print("==================GENERATED OUTPUT===============")
+print('\n'.join(sampled))
+```
+
+
+### Monitoring the execution
+
+RayLLM-Batch uses Ray Data to implement the execution of the batch inference pipeline, and one can use the Ray Dashboard to monitor the execution. In the Ray Dashboard tab, navigate to the Job page and open the "Ray Data Overview" section. Click on the link for the running job, and open the "Ray Data Overview" section to view the details of the batch inference execution:
+
+<img src="https://raw.githubusercontent.com/anyscale/templates/main/templates/batch-llm/assets/ray-data-jobs.png" width=900px />
+
+### Handling GPU out-of-memory failures
+If you run into CUDA out of memory, your batch size is likely too large. Set an explicit small batch size or use a smaller model (or a larger GPU).
```

Comment:
Better to point to the trouble shooting page (we should have that in the future).